src/__tests__/commands/locations.test.ts

```diff
@@ -0,0 +1,135 @@
+import { jest } from '@jest/globals'
+
+import { ArgumentsCamelCase, Argv } from 'yargs'
+
+import { Location, LocationsEndpoint, SmartThingsClient } from '@smartthings/core-sdk'
+
+import { APICommand, APICommandFlags, apiCommand, apiCommandBuilder, apiDocsURL } from '../../lib/command/api-command.js'
+import { outputItemOrList, outputItemOrListBuilder } from '../../lib/command/listing-io.js'
+import { CommandArgs } from '../../commands/locations.js'
+
+
+
+const apiCommandMock: jest.Mock<typeof apiCommand> = jest.fn()
+const apiCommandBuilderMock: jest.Mock<typeof apiCommandBuilder> = jest.fn()
+const apiDocsURLMock: jest.Mock<typeof apiDocsURL> = jest.fn()
+jest.unstable_mockModule('../../lib/command/api-command.js', () => ({
+	apiCommand: apiCommandMock,
+	apiCommandBuilder: apiCommandBuilderMock,
+	apiDocsURL: apiDocsURLMock,
+}))
+
+const outputItemOrListMock: jest.Mock<typeof outputItemOrList> = jest.fn()
+const outputItemOrListBuilderMock: jest.Mock<typeof outputItemOrListBuilder> = jest.fn()
+jest.unstable_mockModule('../../lib/command/listing-io.js', () => ({
+	outputItemOrList: outputItemOrListMock,
+	outputItemOrListBuilder: outputItemOrListBuilderMock,
+}))
+
+
+
+const { default: cmd } = await import('../../commands/locations.js')
+
+
+test('builder', () => {
+	const apiCommandArgvMock = jest.fn() as jest.Mock<Argv<object & APICommandFlags>> & Argv<object & APICommandFlags>
+	apiCommandBuilderMock.mockReturnValue(apiCommandArgvMock)
+
+	const outputItemOrListArgvMock = jest.fn() as jest.Mock<Argv<object & APICommandFlags>> & Argv<object & APICommandFlags>
+
+	const positionalMock = jest.fn() as jest.Mock<Argv<object & APICommandFlags>['positional']> & Argv<object & APICommandFlags>['positional']
+	positionalMock.mockReturnValue(outputItemOrListArgvMock)
+	outputItemOrListArgvMock.positional = positionalMock
+
+	const exampleMock = jest.fn() as jest.Mock<Argv<object & APICommandFlags>['example']> & Argv<object & APICommandFlags>['example']
+	exampleMock.mockReturnValue(outputItemOrListArgvMock)
+	outputItemOrListArgvMock.example = exampleMock
+
+	const epilogMock = jest.fn() as jest.Mock<Argv<object & APICommandFlags>['epilog']> & Argv<object & APICommandFlags>['epilog']
+	epilogMock.mockReturnValue(outputItemOrListArgvMock)
+	outputItemOrListArgvMock.epilog = epilogMock
+
+	outputItemOrListBuilderMock.mockReturnValueOnce(outputItemOrListArgvMock)
+
+	type CommandFlags = APICommandFlags & { testOption?: string }
+	const yargsMock = jest.fn() as jest.Mock<Argv<CommandFlags>> & Argv<CommandFlags>
+
+	const builder = cmd.builder as (yargs: Argv<object>) => Argv<CommandArgs>
+	expect(builder(yargsMock)).toBe(outputItemOrListArgvMock)
+
+	expect(positionalMock).toHaveBeenCalledTimes(1)
+	expect(exampleMock).toHaveBeenCalledTimes(1)
+	expect(epilogMock).toHaveBeenCalledTimes(1)
+})
+
+describe('handler', () => {
```

Comment:
Here and elsewhere: How are we distinguishing between `test()` and `describe()`, if at all?

Reply:
`describe` wraps one or more tests and `test` (or its alias `it`), _is_ a test. What I have been doing is use `test('functionName, ...` when there is only one test to write for a function and `describe` with multiple `it('does this thing', ...` when a function needs more than one test.